core/hashfuncs.h

```diff
@@ -75,9 +76,31 @@ static inline uint32_t hash_djb2_one_float(float p_in,uint32_t p_prev=5381) {
 		uint32_t i;
 	} u;
 
-	// handle -0 case
-	if (p_in==0.0f) u.f=0.0f;
-	else u.f=p_in;
+	// Normalize +/- 0.0 and Nan values so they hash the same.
+	if (p_in==0.0f)
+		u.f=0.0;
+	else if (std::isnan(p_in))
+		u.f=std::numeric_limits<float>::quiet_NaN();
```

Comment:
Can we be sure that this would actually work on consoles and other weirdnesses? (also, I'm not sure we are using std:: anywhere else, so, would be cool if we can get it out somehow)